sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -108,7 +108,9 @@ case class AdaptiveSparkPlanExec(
   @transient private val costEvaluator = SimpleCostEvaluator
 
   @volatile private var currentPhysicalPlan =
-    applyPhysicalRules(initialPlan, queryStagePreparationRules)
+    applyPhysicalRules(inputPlan, queryStagePreparationRules)
+
+  private val initialPlan = currentPhysicalPlan
```

Comment:
nit: It would more intuitive to initialize `intialPlan` first:
```
@transient private val initialPlan =
  applyPhysicalRules(inputPlan, queryStagePreparationRules)

@volatile private var currentPhysicalPlan = initialPlan
```